Provide helpful information when a 'cPanel $version does not appear to be up to date' Blocker is encountered

- **Provide helpful information when a 'cPanel $version does not appear to be up to date' Blocker is encountered**

- This is the Blocker the client sees when their server is not running the most recent version of cPanel v110:

```
......
* 2024-01-08 21:43:45 [INFO] Successfully verified signature for cpanel (key types: release, development).
* 2024-01-08 21:43:45 [ERROR] This installation of cPanel (11.110.0.18) does not appear to be up to date.
  Please upgrade cPanel to a most recent version.
* 2024-01-08 21:43:45 [WARN] Please fix the detected issues before performing the elevation process.
  Read More: https://cpanel.github.io/elevate/blockers/
```

- For v102 the information is the same:

```
......
info [elevate-cpanel] Successfully verified signature for cpanel (key types: release, development).
* 2024-01-08 22:55:59 [ERROR] This installation of cPanel (11.102.0.36) does not appear to be up to date.
  Please upgrade cPanel to a most recent version.
* 2024-01-08 22:55:59 [WARN] Please fix the detected issues before performing the elevation process.
  Read More: https://cpanel.github.io/elevate/blockers/

```

- Maybe, we could tell them what version we are looking for, suggest they make certain updates are enabled in `/etc/cpupdate.conf` and provide a command to perform the upcp, like, If updates are not disabled on your server, then, you may execute the update now using `/usr/local/cpanel/scripts/fix-cpanel-perl; /usr/local/cpanel/scripts/upcp --cron`

--- a/elevate-cpanel
+++ b/elevate-cpanel
@@ -2026,13 +2026,12 @@ EOS
 
     sub _blocker_cpanel_needs_update ($self) {
         if ( !$self->getopt('skip-cpanel-version-check') ) {
-            my $tiers_obj = Cpanel::Update::Tiers->new( logger => Log::Log4perl->get_logger(__PACKAGE__) );
-            if ( !grep { Cpanel::Version::Compare::compare( $Cpanel::Version::Tiny::VERSION_BUILD, '==', $_ ) } $tiers_obj->get_flattened_hash()->@{qw/edge current release stable lts/} ) {
-                my $hint = '';
-                $hint = q[hint: You can skip this check using --skip-cpanel-version-check] if $Cpanel::Version::Tiny::VERSION_BUILD =~ 9999;
+            my $tiers_obj        = Cpanel::Update::Tiers->new( logger => Log::Log4perl->get_logger(__PACKAGE__) );
+            my $expected_version = $tiers_obj->get_flattened_hash()->{'11.110'};
+            if ( !Cpanel::Version::Compare::compare( $Cpanel::Version::Tiny::VERSION_BUILD, '==', $expected_version ) ) {
                 return $self->has_blocker( <<~"EOS" );
             This installation of cPanel ($Cpanel::Version::Tiny::VERSION_BUILD) does not appear to be up to date.
-            Please upgrade cPanel to a more recent version. $hint
+            Please upgrade cPanel to $expected_version.
             EOS
             }
         }

--- a/lib/Elevate/Blockers/WHM.pm
+++ b/lib/Elevate/Blockers/WHM.pm
@@ -107,13 +107,12 @@ sub _blocker_cpanel_needs_license ($self) {
 
 sub _blocker_cpanel_needs_update ($self) {
     if ( !$self->getopt('skip-cpanel-version-check') ) {
-        my $tiers_obj = Cpanel::Update::Tiers->new( logger => Log::Log4perl->get_logger(__PACKAGE__) );
-        if ( !grep { Cpanel::Version::Compare::compare( $Cpanel::Version::Tiny::VERSION_BUILD, '==', $_ ) } $tiers_obj->get_flattened_hash()->@{qw/edge current release stable lts/} ) {
-            my $hint = '';
-            $hint = q[hint: You can skip this check using --skip-cpanel-version-check] if $Cpanel::Version::Tiny::VERSION_BUILD =~ 9999;
+        my $tiers_obj        = Cpanel::Update::Tiers->new( logger => Log::Log4perl->get_logger(__PACKAGE__) );
+        my $expected_version = $tiers_obj->get_flattened_hash()->{'11.110'};
+        if ( !Cpanel::Version::Compare::compare( $Cpanel::Version::Tiny::VERSION_BUILD, '==', $expected_version ) ) {
             return $self->has_blocker( <<~"EOS" );
             This installation of cPanel ($Cpanel::Version::Tiny::VERSION_BUILD) does not appear to be up to date.
-            Please upgrade cPanel to a more recent version. $hint
+            Please upgrade cPanel to $expected_version.
             EOS
         }
     }

--- a/t/blocker-whm.t
+++ b/t/blocker-whm.t
@@ -148,6 +148,8 @@ my $whm  = $cpev->get_blocker('WHM');
     note "cPanel & WHM latest version.";
     clear_messages_seen();
 
+    my $latest_lts_version = "11.110.0.15";
+
     my $mock_tiers = Test::MockModule->new('Cpanel::Update::Tiers');
     $mock_tiers->redefine(
         sync_tiers_file => 1,
@@ -167,6 +169,13 @@ my $whm  = $cpev->get_blocker('WHM');
                         "build"   => "11.102.0.7",
                         "is_main" => 1,
                         "named"   => [ "current", "edge" ],
+                    }
+                ],
+                "11.110" => [
+                    {
+                        "build"   => $latest_lts_version,
+                        "is_main" => 1,
+                        "named"   => [ "lts" ],
                     }
                 ],
             },
@@ -195,7 +204,7 @@ my $whm  = $cpev->get_blocker('WHM');
             msg => qr{
                     \QThis installation of cPanel (11.102.0.5) does not appear to be up to date.\E
                     \s+
-                    \QPlease upgrade cPanel to a more recent version.\E
+                    \QPlease upgrade cPanel to $latest_lts_version.\E
                 }xms,
         },
         q{obsolete version generates a blocker.}
@@ -203,7 +212,7 @@ my $whm  = $cpev->get_blocker('WHM');
 
     clear_messages_seen();
 
-    $Cpanel::Version::Tiny::VERSION_BUILD = '11.102.0.7';
+    $Cpanel::Version::Tiny::VERSION_BUILD = $latest_lts_version;
     is( $whm->_blocker_cpanel_needs_update(), 0, "No blocker if cPanel is up to date" );
     no_messages_seen();
 }

